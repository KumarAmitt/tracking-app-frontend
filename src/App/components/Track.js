import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import AppBar from './AppBar';
import { getAllDeals, loadDeals } from '../../store/slicers/deal';
import { getProducts, loadProducts } from '../../store/slicers/product';

const Track = () => {
  const title = 'Track Your Deal';
  const dispatch = useDispatch();
  const deals = useSelector(getAllDeals);
  const products = useSelector(getProducts);

  useEffect(() => {
    dispatch(loadDeals());
    dispatch(loadProducts());
  }, []);

  console.log(products);
  console.log(deals);

  const handleClick = (data) => {
    console.log(data);
  };

  // products.length > 0 ? products.filter((f) => f.id === e.product_id)[0].product_name : ''

  return (
    <>
      <AppBar title={title} />
      {
        Object.entries(deals).map((d) => (
          <div key={d[0]} onClick={() => handleClick(d[1])} onKeyDown={() => handleClick(d[1])} role="button" tabIndex={0}>
            <span>
              {d[0]}
              :
            </span>
            {' '}
            <span>{d[1].map((e) => e.premium).reduce((a, b) => a + b)}</span>
            {' '}
            <span>{d[1].length}</span>
          </div>
        ))
      }

      {
        Object.entries(deals).map((d) => (
          <Link to="/track_details" key={d[0]}>Deal Details</Link>
        ))
      }

    </>
  );
};

export default Track;
