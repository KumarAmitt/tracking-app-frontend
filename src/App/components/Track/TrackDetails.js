import React, { useEffect } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import AppBar from '../AppBar/AppBar';
import { getProducts, loadProducts } from '../../../store/slicers/product';
import './style/TrackDetails.css';
import Stats from './Stats';
import { getUserInfo, loadSession } from '../../../store/slicers/user';

const TrackDetails = ({ location }) => {
  const { info, date } = location;
  const title = 'Track Details';
  const dispatch = useDispatch();
  const userInfo = useSelector(getUserInfo);
  const products = useSelector(getProducts);
  const todayTotal = info ? info.map((e) => e.premium).reduce((a, b) => a + b) : 0;

  useEffect(() => {
    dispatch(loadSession());
    dispatch(loadProducts());
  }, []);

  if (!userInfo.logged_in) {
    return <Redirect to="/track" />;
  }

  return (
    <>
      <AppBar title={title} link="/track" />
      <div className="details">
        <div className="details-header">
          <div className="date">{date}</div>
          <Stats today={todayTotal || 0} />
        </div>
        <div className="details-items">
          {
            info && info.map((e) => (
              <div key={e.id} className="details-item">
                <div className="ins-type">{products.length > 0 ? products.filter((f) => f.id === e.product_id)[0].product_name : ''}</div>
                <div className="premium">
                  <span>&#8377;</span>
                  {' '}
                  {e.premium}
                </div>
                <div className="appl-id">{e.application_id}</div>
              </div>
            ))
          }
        </div>
      </div>
    </>
  );
};

TrackDetails.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default TrackDetails;
