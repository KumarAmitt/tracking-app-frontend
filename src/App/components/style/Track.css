.track {
    background-color: #eee;
    height: 100vh;
}
.not-found {
    font-size: 1.1rem;
    margin-top: 1em;
    text-align: center;
    color: var(--accent-primary);
    font-weight: bold;
}

.track-tile {
    margin: 0.8em 0;
    padding: 0.4em 0.8em;
    background-color: #fff;
}

.track-tile a {
    text-decoration: none;
    display: flex;
    justify-content: space-between;
    align-items: center;
    color: var(--accent-secondary);
    transition: all linear 300ms;
}

.track-tile a:hover {
    transform: scaleX(1.01);
}

.track-tile-left .date {
    font-size: 1.2rem;
    font-weight: bold;
}

.track-tile-left .count {
    color: lightslategray;
}

.track-tile-right {
    font-weight: bold;
    font-size: 1.1rem;
    display: flex;
}

.track-tile-right > * {
    margin-left: 0.4em;
}

.track-tile-right span {
    margin-left: 0.4em;
    margin-top: 0.1em;
}