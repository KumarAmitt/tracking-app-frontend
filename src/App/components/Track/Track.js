import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';
import NavigateNextIcon from '@material-ui/icons/NavigateNext';
import AppBar from '../AppBar/AppBar';
import { getAllDeals, loadDeals } from '../../../store/slicers/deal';
import './style/Track.css';
import { getUserInfo, loadSession } from '../../../store/slicers/user';

const Track = () => {
  const title = 'Track Your Deal';
  const dispatch = useDispatch();

  const userInfo = useSelector(getUserInfo);
  const deals = useSelector(getAllDeals);

  useEffect(() => {
    dispatch(loadSession());
    dispatch(loadDeals());
  }, []);

  if (userInfo.logged_in === false) {
    return <Redirect to="/" />;
  }

  return (
    <div className="track">
      <AppBar title={title} />
      {
        Object.entries(deals).length === 0 ? <div className="not-found">No data found. Please Add deal first</div>
          : Object.entries(deals).map((d) => (
            <div key={d[0]} className="track-tile">
              <Link to={{ pathname: '/track_details', info: d[1], date: d[0] }}>
                <div className="track-tile-left">
                  <div className="date">{d[0]}</div>
                  <div className="count">
                    Policy Count:
                    {' '}
                    <span>{d[1].length}</span>
                  </div>
                </div>
                <div className="track-tile-right">
                  <span>&#8377;</span>
                  <span>{d[1].map((e) => e.premium).reduce((a, b) => a + b)}</span>
                  <NavigateNextIcon />
                </div>
              </Link>
            </div>
          ))
       }
    </div>
  );
};

export default Track;
