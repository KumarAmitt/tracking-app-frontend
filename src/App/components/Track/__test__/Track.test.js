import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import Track from '../Track';

const store = configureAppStore();

describe('Track', () => {
  it('should match the snapshot of Track component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Track />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
