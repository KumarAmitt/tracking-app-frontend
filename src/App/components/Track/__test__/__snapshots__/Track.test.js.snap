// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track should match the snapshot of Track component 1`] = `null`;
